core/ecschema-editing/src/Merging/ClassMerger.ts

```diff
@@ -2,209 +2,187 @@
+    return baseClassSetter(item.key, baseClassKey);
+  }
+  if(item.baseClass !== undefined) {
+    const currentBaseClass = await item.baseClass;
+    const newBaseClass = await context.editor.schemaContext.getSchemaItem<ECClass>(baseClassKey);
+    if (newBaseClass === undefined) {
+      return { errorMessage: `'${baseClassKey.name}' class could not be located in the merged schema.`};
+    }
+    if(await newBaseClass.is(currentBaseClass)) {
+      return baseClassSetter(item.key, baseClassKey);
     }
   }
+  return { errorMessage: `Changing the class '${item.key.name}' baseClass is not supported.`};
+}
 
-  // 2nd pass to merge baseClass, properties, mixins and CA.
-  public static async mergeItemContentChanges(context: SchemaMergeContext, classChanges: Iterable<ClassChanges>){
-    const merger = new this(context);
-    for (const change of classChanges) {
-      const targetItemKey = new SchemaItemKey(change.ecTypeName, context.targetSchema.schemaKey);
-      const changeType = change.schemaItemMissing?.changeType;
-
-      if (change.baseClassDelta !== undefined) {
-        await merger.handleError(merger.setBaseClass(targetItemKey, change.baseClassDelta, changeType));
-      }
-
-      // merge class attributes
-      await merger.handleError(merger.mergeAttributeValueChanges(targetItemKey, change.propertyValueChanges));
-
-      // merge class mixins/constraints/etc
-      await merger.handleError(merger.merge(targetItemKey, change));
-
-      // merge class property attribute values
-      if (change.propertyChanges.size > 0) {
-        await merger.handleError(merger.mergePropertyChanges(targetItemKey, change.propertyChanges.values()));
-      }
+async function setClassModifier(item: MutableClass, modifierValue: string): Promise<SchemaEditResults> {
+  const modifier = parseClassModifier(modifierValue);
+  if(modifier === undefined) {
+    return { errorMessage: "An invalid class modifier has been provided." };
+  }
+  if(item.modifier === undefined || item.modifier === modifier || modifier === ECClassModifier.None) {
+    item.setModifier(modifier);
+    return {};
+  }
+  return { errorMessage: `Changing the class '${item.name}' modifier is not supported.` };
+}
 
-      // merge custom attributes
-      await merger.handleError(mergeCustomAttributes(merger.context, change.customAttributeChanges.values(), async (ca) => {
-        return merger.context.editor.entities.addCustomAttribute(targetItemKey, ca);
-      }));
+function getBaseClassSetter(context: SchemaMergeContext, item: ECClass) {
+  return async (itemKey: SchemaItemKey, baseClassKey: SchemaItemKey) => {
+    switch(item.schemaItemType) {
+      case SchemaItemType.EntityClass: return context.editor.entities.setBaseClass(itemKey, baseClassKey);
+      case SchemaItemType.Mixin: return context.editor.mixins.setMixinBaseClass(itemKey, baseClassKey);
+      // TODO: verify; structs and relationship classes can't have base classes?
```

Comment:
Relationships very often have base classes.  Technically structs can have base classes but it is discouraged.  So we should probably support both here.

Reply:
I've created a backklog item to cover that #6582